metals/src/main/scala/scala/meta/internal/metals/FindTextInFiles.scala

```diff
@@ -0,0 +1,130 @@
+package scala.meta.internal.metals
+
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.Location
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+
+class FindTextInFiles(
+    buildTargets: BuildTargets,
+    workspace: () => AbsolutePath
+) {
+  def find(mask: String, content: String): List[Location] = {
+    if (mask == null || content == null) {
+      List.empty[Location]
+    } else {
+      val allLocations: ArrayBuffer[Location] = new ArrayBuffer[Location]()
+
+      buildTargets.allWorkspaceJars.foreach { classpathEntry =>
+        try {
+          val jarLocations: List[Location] =
+            if (classpathEntry.isFile && classpathEntry.isJar) {
+              visitJar(classpathEntry, mask, content)
+            } else Nil
+
+          allLocations ++= jarLocations
+        } catch {
+          case NonFatal(e) =>
+            scribe.error(
+              s"Failed to find in non-source files for $classpathEntry",
+              e
+            )
+        }
+      }
+
+      allLocations.toList
+    }
+  }
+
+  def isSuitableFile(path: AbsolutePath, mask: String): Boolean = {
+    path.isFile && path.filename.contains(mask)
+  }
+
+  def visitJar(
+      path: AbsolutePath,
+      mask: String,
+      content: String
+  ): List[Location] = {
+    val jarLocations: ArrayBuffer[Location] = new ArrayBuffer[Location]()
+
+    FileIO.withJarFileSystem(path, create = false, close = true) { root =>
+      Files.walkFileTree(
+        root.toNIO,
+        new SimpleFileVisitor[Path] {
+          override def visitFile(
+              file: Path,
+              attrs: BasicFileAttributes
+          ): FileVisitResult = {
+            val absPath = AbsolutePath(file)
+            if (isSuitableFile(absPath, mask)) {
+              val fileRanges: List[Range] = visitFileInsideJar(absPath, content)
+
+              val fileLocations: List[Location] =
+                if (fileRanges.nonEmpty) {
+                  val result = absPath.toFileOnDisk(workspace())
+                  fileRanges.map(range => new Location(result.toString, range))
```

Comment:
Location.uri takes a string but it's actually should be an uri encoded string
```suggestion
                  fileRanges.map(range => new Location(result.toURI.toString, range))
```

Reply:
done